auth/hostregistry.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2016 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// maintains a map of authenticated hosts and their
+// authentication token expiration time
+
+package auth
+
+import (
+	"errors"
+	"sync"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+var (
+	// ErrMissingHost indicates if a host does not
+	// exist in the host expiration registry
+	ErrMissingHost = errors.New("Host is not present in host expiration registry")
+)
+
+// HostExpirationRegistry is a threadsafe map of
+// host id to auth expiration time. NOTE: expired hosts
+// are not removed from the registry
+type HostExpirationRegistry struct {
+	registry map[string]int64
+	sync.Mutex
```

Comment:
Please use a `sync.RWMutex` instead. It'll add significant performance benefits in the `IsExpired` call.